Show notice for non existent controller property.

ADmad opened the pull request:
Refs https://github.com/cakephp/cakephp/issues/12590#issuecomment-423932380

ADmad committed:
Show notice for non existent controller property.

garas commented on src/Controller/Controller.php:
```diff
@@ -371,6 +371,19 @@ public function __get($name)
 
         list($plugin, $class) = pluginSplit($this->modelClass, true);
         if ($class !== $name) {
```
Maybe reverse `if` and explain:

```
if ($class === $name) {
   // Autoload table if $name matches controller name
   return $this->loadModel($plugin . $class);
}

// notice

return false; // remove or explicitly return null at 4.x
```

ADmad committed:
Improve code readability.

markstory commented:
Looks like we have a few undefined properties being used by our tests :cry:

ADmad commented:
Not really, the error doesn't make sense. It's complaining about missing `Controller::$viewClass` but that property is available through the `ViewVarsTrait`.

ADmad committed:
Fix test.

garas commented:
`Controller::$viewClass` is defined in `ViewVarsTrait`, but it is `unset()` after first use.

> https://github.com/cakephp/cakephp/pull/12035 I've added an unset() for viewClass to avoid a permanent deprecation warning when RequestHandlerComponent is used.

https://github.com/cakephp/cakephp/blob/13ab4d788ce553056938b60b3039197b8edd9993/src/View/ViewVarsTrait.php#L75-L78

If second `render()` tries access value, it is no more available. It should probably leave property set.

ADmad committed:
Set property to null instead of unsetting.

This avoid triggering Controller::__get() on subsequent access to ViewVarsTrait::$viewClass.

ADmad commented:
`PHPUnit\Framework\Error\Notice` is not available in phpunit 5.7 :slightly_frowning_face:

ADmad committed:
Fix test for phpunit 5.7.

ADmad commented:
Meh, took care of it with a `class_exists()` check.
